serde/src/private/de.rs

```diff
@@ -2587,17 +2587,43 @@ where
 }
 
 #[cfg(any(feature = "std", feature = "alloc"))]
-pub struct FlatMapDeserializer<'a, 'de: 'a, E>(
-    pub &'a mut Vec<Option<(Content<'de>, Content<'de>)>>,
-    pub PhantomData<E>,
-);
+#[cfg_attr(any(feature = "std", feature = "alloc"), derive(Debug))]
```

Comment:
Conditional derive seems not to be necessary because the whole struct is under that condition.